datafusion/core/src/datasource/view.rs

```diff
@@ -0,0 +1,364 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! View data source which uses a LogicalPlan as it's input.
+
+use std::{any::Any, sync::Arc};
+
+use arrow::datatypes::SchemaRef;
+use async_trait::async_trait;
+
+use crate::{
+    error::Result,
+    execution::context::SessionContext,
+    logical_plan::{Expr, LogicalPlan},
+    physical_plan::ExecutionPlan,
+};
+
+use crate::datasource::{TableProvider, TableType};
+
+/// An implementation of `TableProvider` that uses another logical plan.
+pub struct ViewTable {
+    /// To create ExecutionPlan
+    context: SessionContext,
+    /// LogicalPlan of the view
+    logical_plan: LogicalPlan,
```

Comment:
Original SQL might be useful too to store, to preserve formatting when doing things like `describe view`.

Reply:
Indeed that would be nice, do you think we could do it as a follow on PR?

Reply:
I think so

Reply:
hi, I'm new to datafusion and maybe I can help with the follow work ?

Reply:
👋  welcome @Veeupup . That would be great.  I will try and write up a ticket later today that describes the work in some more detail in case that is helpful

Thank you!

Reply:
https://github.com/apache/arrow-datafusion/issues/2529

@Dandandan  not sure if you had specific ideas on what command you wanted to show view definitions.